_episodes/05-preprocessor.md

```diff
@@ -0,0 +1,287 @@
+>>       mip: Amon
+>>       grid: gn #can change for variables from the same model
+>>       start_year: 1970 
+>>       end_year: 2000
+>>       additional_datasets:
+>>     scripts: null
+>> ~~~
+>> {: .source}
+> {: .solution}
+{: .challenge}
+
+## Adding different datasets for different variables
+
+Sometimes, we may want to include specific datasets for certain variables. An example is when we use observations for two different variables in a diagnostic. While the CMIP dataset details for the two variables may be common, the observations will likely not be so. It would be useful to know how to include different datasets for different variables. Here is an example of a simple preprocessor and diagnostic setup for that:
+
+> ## Example
+>~~~YAML
+>
+> datasets:
+>   - {dataset: UKESM1-0-LL, project: CMIP6, exp: historical, 
+>      ensemble: r1i1p1f2} #common to both variables discussed below
+> 
+> preprocessors:
+>   prep_regrid: # regrid to get all data to the same resolution
+>     regrid:    #apply the preprocessor to regrid
+>       target_grid: 2.5x2.5 # target resolution
+>       scheme: linear  #how to interpolate for regridding
+> 
+> diagnostics:
+>   # --------------------------------------------------      
+>   # Simple diagnostic to illustrate use of different
+>   # datasets for different variables
+>   # --------------------------------------------------
+>   diag_diff_datasets:
+>     description: diff_datasets_for_vars
+>     variables:
+>       pr:  #first variable is precipitation
+>         preprocessor: prep_regrid
+>         mip: Amon
+>         grid: gn #can change for variables from the same model
+>         start_year: 1970  
+>         end_year: 2000 #  start and end years for a30 year period, 
+>                        # we assume this is common and exists for all
+>                        # model and obs data 
+>         additional_datasets:
+>           - {dataset: GPCP-SG, project: obs4mips, level: L3, 
+>              version: v2.2, tier: 1} #dataset specific to this variable
+> 
+>       tas: #second variable is surface temperature
+>         preprocessor: prep_regrid
+>         mip: Amon
+>         grid: gn #can change for variables from the same model
+>         start_year: 1970  #some 30 year period
+>         end_year: 2000
+>         additional_datasets:
+>           - {dataset: HadCRUT4, project: OBS, type: ground, 
+>              version: 1, tier: 2} #dataset specific to the temperature var
+> 
+>     scripts: null
+>~~~
+>{: .source}
+{: .solution}
+
+> ## How to find what CMIP data is available?
+>
+> [CMIP5](https://pcmdi.llnl.gov/mips/cmip5/index.html) and [CMIP6](https://pcmdi.llnl.gov/CMIP6/Guide/dataUsers.html) data obey the [CF-conventions](http://cfconventions.org/). Available variables could be found under the [CMIP5 data request](https://pcmdi.llnl.gov/mips/cmip5/docs/standard_output.pdf?id=28) and the [CMIP6 Data Request](http://clipc-services.ceda.ac.uk/dreq/index.html).
```

Comment:
Should be rephrased to "Available variables can be found under the ..."? "Could" sounds here so speculative...